hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/NameNode.java

```diff
@@ -1899,6 +1899,10 @@ synchronized void transitionToStandby() throws IOException {
   synchronized void transitionToObserver() throws IOException {
     String operationName = "transitionToObserver";
     namesystem.checkSuperuserPrivilege(operationName);
+    if (namesystem.isInSafeMode()) {
```

Comment:
I think we can guard this by `notBecomeActiveInSafemode`. Though the config claims to be about "active" status, the logic in `monitorHealth` just generally considers a standby NN as unhealthy if it's in safemode, and I think the intent here is the same as with that config.


Also: `namesystem.isInSafeMode()` -> `isInSafeMode()` (`NameNode` redefines this method)

Reply:
Thank you @xkrogen very much for the review and careful suggestions. 

The intent here is really the same as `dfs.ha.nn.not-become-active-in-safemode`. It's just that the configuration name looks a bit conflicting.